src/test/pages/Gameboard.test.tsx

```diff
@@ -0,0 +1,144 @@
+import { Gameboard } from "../../app/components/pages/Gameboard";
+import { renderTestEnvironment } from "../utils";
+import { screen, waitFor, within } from "@testing-library/react";
+import { Role } from "../../IsaacApiTypes";
+import { rest } from "msw";
+import { API_PATH, USER_ROLES } from "../../app/services";
+import { mockGameboards } from "../../mocks/data";
+import { produce } from "immer";
+import userEvent from "@testing-library/user-event";
+import { isaacApi } from "../../app/state";
+
+const renderGameboard = ({
+  role,
+  notGameboardOwner,
+  notSavedToCurrentUser,
+}: {
+  role: Role;
+  notGameboardOwner?: boolean;
+  notSavedToCurrentUser?: boolean;
+}) => {
+  const mockGameboard = {
+    ...mockGameboards.results[0],
+    ownerUserId: notGameboardOwner ? 2 : 301,
+    savedToCurrentUser: !notSavedToCurrentUser,
+  };
+
+  renderTestEnvironment({
+    modifyUser: (user) =>
+      produce(user, (u) => {
+        u.role = role;
+        u.id = 301;
+      }),
+    PageComponent: Gameboard,
+    initialRouteEntries: ["/gameboard#test-gameboard-1"],
+    extraEndpoints: [
+      rest.get(API_PATH + "/gameboards/test-gameboard-1", (req, res, ctx) => {
+        return res(ctx.status(200), ctx.json(mockGameboard));
+      }),
+      rest.post(API_PATH + "/gameboards/test-gameboard-1", (req, res, ctx) => {
+        return res(ctx.status(200), ctx.json({ ...mockGameboard, title: req.url.searchParams.get("title") }));
+      }),
+    ],
+  });
+  return mockGameboard;
+};
+
+describe("Gameboard", () => {
+  it("renders expected title", async () => {
+    const gameboard = renderGameboard({ role: "STUDENT" });
+    await waitFor(() => {
```

Comment:
It looks like you are using this same block in all the tests, so it may be worth moving it to its own function (in this file, maybe), with a descriptive name, so that it's a bit more readable. I think that the intention is to wait for the loader to disappear, so any name that captures that would be fine.

Reply:
Alternatively, you could move it inside `renderGameboard`, if you think that waiting for the loader to disappear can be considered the last step in the rendering process (which is a bit debatable).

Reply:
Moved inside `renderGameboard`